src/Client.php

```diff
@@ -301,6 +323,28 @@ protected function getContext(ConnectionInterface $connection)
     }
 
     /**
+     * Derives a set of socket context options for a given secure connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return array Associative array of context option key-value pairs
+     */
+    protected function getSecureContext(ConnectionInterface $connection)
+    {
+        $context = array();
+        if ($this->getForceIpv4Flag($connection)) {
+            $context['bindto'] = '0.0.0.0:0';
+        }
+        if ($this->getVerifyPeerFlag($connection)) {
+            $context['verify_peer'] = false;
+        }
+        if ($this->getAllowSelfSignedFlag($connection)) {
+            $context['allow_self_signed'] = true;
+        }
+        //$context = array('socket' => $context);
```

Comment:
I think removing this line, by commenting it out or otherwise, makes this code incorrect. See [these examples](http://php.net/manual/en/context.socket.php#refsect1-context.socket-examples) of using socket context options.